[New] External Secret Operator with Secret Manager (#8653)
ovh/docs: pages/manage_and_operate/secret_manager/external-secret-operator/guide.en-gb.md

---
title: "Use Kubernetes External Secret Operator with Secret Manager"
excerpt: "Configure External Secret Operator to store Kubernetes secrets on the OVHcloud Secret Manager"
updated: 2025-11-07
---

> [!primary]
> Secret Manager is currently in beta phase. This guide can be updated in the future with the advances of our teams in charge of this product.

## Objective

This guide explains how to set up the Kubernetes External Secret Operator to use the OVHcloud Secret Manager as a provider.

## Requirements

- An [OVHcloud customer account](/pages/account_and_service_management/account_information/ovhcloud-account-creation).
- Have [ordered an OKMS domain](/pages/manage_and_operate/kms/quick-start) or [created a first secret](/pages/manage_and_operate/secret_manager/secret-manager-ui).
- Have a Kubernetes cluster.

## Instructions

### Setup the Secret Manager

To allow access to the Secret Manager you will need to create credentials.

> Create an [IAM local user](/pages/account_and_service_management/account_information/ovhcloud-users-management) with access right on your domain.

Reviewer: tant que l'on a pas une CLI, rajouter des captures ecrans pour montrer toutes les etapes de la creation de l'utilisateur local avec les droits qui vont bien.

Reviewer: On peut maintenant créer un user avec la CLI :

```
$ ovhcloud iam user create --login "secretmanager-b1033fdd-xxxx-xxxx-xxxx-xxxxxxxxx" --group ADMIN --description "A user create for secretmanager, linked to xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx" --password "secretmanager-xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx" --email "secretmanager-xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx@ovhcloud.com"
```

The user should be a member of a group with the ADMIN role, or if using [IAM policies](/pages/account_and_service_management/account_information/iam-policy-ui) to have at least the following rights on the OKMS domain:

- `okms:apikms:secret/create`
- `okms:apikms:secret/version/getData`
- `okms:apiovh:secret/get`

Then create a Personnal Acces Token (PAT) `user_pat`:

> [!api]
>
> @api {v1} /me POST /me/identity/user/{user}/token

API will answer with:

```json
{
  "creation": "2025-11-13T10:38:44.658926311Z",
  "description": "my first PAT",
  "expiresAt": null,
  "lastUsed": null,
  "name": "my_PAT",
  "token": "eyJhbGciOiJFZERXXXXXXXXDyI23Q7euIDmw9Pn__SDA"
}
```

Keep safe the value of `token` field as it will never be prompt again and will be used to authenticate on the Secret Manager as `user_pat`.

> You will also need the `okms-id` of the OKMS domain you want to use. This ID can be found on the OVHcloud Control Panel.

Reviewer: recuperable avec la CLI ovhcloud :) 

```
$ ovhcloud okms list
┌──────────────────────────────────────┬─────────────┐
│                  id                  │   region    │
├──────────────────────────────────────┼─────────────┤
│ xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx │ eu-west-par │
│ xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx │ eu-west-par │
└──────────────────────────────────────┴─────────────┘
```

> You will also need the `okms-id` of the OKMS domain you want to use. This ID can be found on the OVHcloud Control Panel.

Reviewer: ```suggestion
You will also need the `region` and the `okms-id` of the OKMS domain you want to use. This ID and this region can be found on the OVHcloud Control Panel.

Or through the `ovhcloud` CLI:

```
$ ovhcloud okms list
┌──────────────────────────────────────┬─────────────┐
│                  id                  │   region    │
├──────────────────────────────────────┼─────────────┤
│ xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx │ eu-west-par │
│ xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx │ eu-west-par │
└──────────────────────────────────────┴─────────────┘
```
```

> ### Setup Sealed Secret (optionnal)

Reviewer: pas necessaire, montrer comment faire sans creation d'un secret)

Sealed Secret allows you to safely store Kubernetes Secrets wherever you want by encrypting them.
This step is optionnal but highly recommended.

> First, install the controller in your cluster.  It will automatically decrypt Sealed Secrets into standard Kubernetes Secrets

Reviewer: ```suggestion
First, install the controller in your cluster. It will automatically decrypt Sealed Secrets into standard Kubernetes Secrets
```

```bash
helm repo add sealed-secrets https://bitnami-labs.github.io/sealed-secrets
helm install sealed-secrets -n kube-system sealed-secrets/sealed-secrets
```

Then, install kubeseal cli to encrypt Secrets into Sealed Secrets

```bash

> KUBESEAL_VERSION='' # Set this to, for example, KUBESEAL_VERSION='0.23.0'

Reviewer: 0.33.1 est la version que je viens d'installer, la plus recente.
Le readme dont est extrait ce code est pas super a jour, donc plutot : 

```
KUBESEAL_VERSION=$(curl -s https://api.github.com/repos/bitnami-labs/sealed-secrets/tags | jq -r '.[0].name' | cut -c 2-)
```

curl -OL "https://github.com/bitnami-labs/sealed-secrets/releases/download/v${KUBESEAL_VERSION:?}/kubeseal-${KUBESEAL_VERSION:?}-linux-amd64.tar.gz"
tar -xvzf kubeseal-${KUBESEAL_VERSION:?}-linux-amd64.tar.gz kubeseal
sudo install -m 755 kubeseal /usr/local/bin/kubeseal

> ```

Reviewer: or install via homebrew

brew install kubeseal

More information: (<https://github.com/bitnami-labs/sealed-secrets>)

### Setup the Secret Provider in Kubernetes

#### Install the External Secret Operator on your kubernetes

```bash
helm repo add external-secrets https://charts.external-secrets.io
helm repo update

helm install external-secrets \
external-secrets/external-secrets \
-n external-secrets \

> --create-namespace \

Reviewer: ```suggestion
--create-namespace
```

```

#### Configure External Secret Operator

> First, setup a `SecretStore` that is responsible of the synchronization with the Secret Manager.

Reviewer 2: `ClusterSecretStore` instead, the SecretStore doesn't work. 
May change all appearance of it.

We configure the ClusterSecretStore using HashiCorp Vault with token authentification and with the OKMS endpoint as backend.

> Add the `user_pat` as a secret to be able to use it in the charts.

Reviewer: cette etape est à mettre apres installation de sealed-secret, dans la section "Setup Sealed Secret (optionnal)"

```yaml

> ---

Reviewer: ```suggestion
```

apiVersion: bitnami.com/v1alpha1
kind: SealedSecret
metadata:
  name: token-secret
  namespace: default
spec:
  encryptedData:
      token: <user_pat>
  template:
      metadata:
        name: token-secret
        namespace: default
        type: Opaque
```

> 

Reviewer: Create the SealedSecret in your cluster:

```
kubectl apply -f sealedsecret.yaml
```

> The `ClusterSecretStore` resource:

Reviewer: ```suggestion
To define a new `ClusterSecretStore` resource, create a `clustersecretstore.yaml` file with the follwoing content:
```

```yaml
apiVersion: external-secrets.io/v1
kind: ClusterSecretStore
metadata:
  name: vault-secret-store
spec:
  provider:
      vault:

>         server: "https://{region}.okms.ovh.net/api/<okms_id>" # OKMS endpoint, fill with the correct region and your okms_id 

Reviewer: ```suggestion
        server: "https://<region>.okms.ovh.net/api/<okms_id>" # OKMS endpoint, fill with the correct region and your okms_id 
```

        path: "secret"
        version: "v2" 
        auth:
            tokenSecretRef:
              name: token-secret # The k8s secret that contain your PAT
              key: token 
```

> [!info]
> Only [token authentication](https://external-secrets.io/latest/provider/hashicorp-vault/#token-based-authentication) is supported

Region name can be translated from your region location using:

> [!api]
>
> @api {v1} /location GET /location

As an example for **Europe (France - Paris)**, OKMS endpoint is **eu-west-par.okms.ovh.net**

#### Use External Secret Operator

Once the `ClusterSecretStore` is setup you can define `ExternalSecret` that comes from the secret manager.

> In the example we use a secret already created on the Secret Manager:

Reviewer: ```suggestion
In the example we use a secret already created on the Secret Manager:
```

- Path: `prod/database/MySQL`
- Value:
  - `login: admin`
  - `password: my_secret_password`

> ```yaml

Reviewer: ```suggestion
Create a `externalsecret.yaml` file with this content:

```yaml
```

apiVersion: external-secrets.io/v1
kind: ExternalSecret
metadata:
  name: vault-external-secret
  namespace: default
spec:
  secretStoreRef:
    name: vault-secret-store
    kind: ClusterSecretStore
  refreshInterval: "10s"
  target:
    name: creds-secret
    creationPolicy: Owner
  data:
    - secretKey: login
      remoteRef:
        key: prod/database/MySQL # Path of the secret in the Secret Manager
        property: login # Key to find in the JSON data of the secret
    - secretKey: password
      remoteRef:
        key: prod/database/MySQL
        property: password
```

> [!info]

> > Only `ExternalSecret` are supported yet.

Reviewer: c'est a dire ?

> 

Reviewer: ```suggestion

Apply the resource in your cluster:

kubectl apply -f externalsecret.yaml

```

#### Deploy your application

> The secret should be created and available in kubernetes.

Reviewer: ```suggestion
The secret should be created and available in the Kubernetes cluster.

Check:

$ kubectl get secret -n default
NAME                   TYPE                             DATA   AGE
token-secret   Opaque                           1      17h
creds-secret        Opaque   1      9m4s
```

---
title: "Use Kubernetes External Secret Operator with Secret Manager"
excerpt: "Configure External Secret Operator to store Kubernetes secrets on the OVHcloud Secret Manager"
updated: 2025-11-07
---

> [!primary]

> > Secret Manager is currently in beta phase. This guide can be updated in the future with the advances of our teams in charge of this product.

Reviewer: ```suggestion
> Secret Manager is currently in Beta phase. This guide can be updated in the future with the advancements made by our teams in charge of this product.
```

## Objective

This guide explains how to set up the Kubernetes External Secret Operator to use the OVHcloud Secret Manager as a provider.

## Requirements

- An [OVHcloud customer account](/pages/account_and_service_management/account_information/ovhcloud-account-creation).
- Have [ordered an OKMS domain](/pages/manage_and_operate/kms/quick-start) or [created a first secret](/pages/manage_and_operate/secret_manager/secret-manager-ui).
- Have a Kubernetes cluster.

## Instructions

### Setup the Secret Manager

> To allow access to the Secret Manager you will need to have a `token`, and the `region` and `okms-id` of your Secret Manager.

Reviewer: ```suggestion
To allow access to the Secret Manager you will need to have a `token`, the `region` and `okms-id` of your Secret Manager.
```

The user should be a member of a group with the ADMIN role, or if using [IAM policies](/pages/account_and_service_management/account_information/iam-policy-ui) to have at least the following rights on the OKMS domain:

- `okms:apikms:secret/create`
- `okms:apikms:secret/version/getData`
- `okms:apiovh:secret/get`

Then create a Personnal Acces Token (PAT) `user_pat`:

> [!tabs]
> API
>> > [!api]
>> >
>> > @api {v1} /me POST /me/identity/user/{user}/token
>>
>> With the following payload (fill with your values):
>>
>> ```json
>> {
>>   "description": "PAT secret manager for domain xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx",
>>   "name": "pat-secretmanager-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx"
>> }
>> ```
>>
>> API will answer with:
>>
>> ```json
>> {
>>   "creation": "2025-11-13T10:38:44.658926311Z",
>>   "description": "PAT secret manager for domain xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx",
>>   "expiresAt": null,
>>   "lastUsed": null,
>>   "name": "pat-secretmanager-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx",
>>   "token": "eyJhbGciOiJ...punpVAg"
>> }
>> ```
>>
> CLI

> >> PAT can also created with the [OVHcloud CLI](https://github.com/ovh/ovhcloud-cli) and the command (fill with your values) :

Reviewer: ```suggestion
>> PAT can also created with the [OVHcloud CLI](https://github.com/ovh/ovhcloud-cli) and the command (fill with your values):
```

>>
>> ```bash

> >> ovhcloud iam user {user} token create --name pat-secretmanager-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx --description "PAT secret manager for domain xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx"

Reviewer: ```suggestion
>> ovhcloud iam user token create {user} --name pat-secretmanager-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx --description "PAT secret manager for domain xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx"
```

>> ✅ Token Secret-Manager created successfully, value: eyJhbGciOiJ...punpVAg
>> ```

Keep safe the value of `token` field as it will never be prompt again and will be used to authenticate on the Secret Manager as `user_pat`.

#### Secret Manager info

You will also need the `region` and the `okms-id` of the OKMS domain you want to use. This ID and this region can be found on the OVHcloud Control Panel.

Or through the [`ovhcloud` CLI](https://github.com/ovh/ovhcloud-cli):

```bash
$ ovhcloud okms list
┌──────────────────────────────────────┬─────────────┐
│ id                                   │ region      │
├──────────────────────────────────────┼─────────────┤
│ xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx │ eu-west-par │
│ xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx │ eu-west-par │
└──────────────────────────────────────┴─────────────┘
```

### Setup the Secret Provider in Kubernetes

#### Install the External Secret Operator on your kubernetes

```bash
helm repo add external-secrets https://charts.external-secrets.io
helm repo update

helm install external-secrets \
   external-secrets/external-secrets \
    -n external-secrets \
    --create-namespace \
    --set installCRDs=true
```

> Check ESO is running :

Reviewer: ```suggestion
Check ESO is running:
```

NAME                                                          DESIRED   CURRENT   READY   AGE
replicaset.apps/external-secrets-8cbc56569                    1         1         1       13s
replicaset.apps/external-secrets-cert-controller-565fcd479b   1         1         0       13s
replicaset.apps/external-secrets-webhook-7fb59d4b88           1         1         0       13s
```

#### Configure External Secret Operator

First, setup a `SecretStore` that is responsible of the synchronization with the Secret Manager.
We configure the SecretStore using HashiCorp Vault with token authentification and with the OKMS endpoint as backend.

Add the `user_pat` as a secret to be able to use it in the charts.

To define a new `SecretStore` resource, create a `secretstore.yaml` file with the followong content:

```yaml
apiVersion: external-secrets.io/v1
kind: SecretStore
metadata:
  name: vault-secret-store
spec:
  provider:
      vault:
        server: "https://<region>.okms.ovh.net/api/<okms_id>" # OKMS endpoint, fill with the correct region and your okms_id 
        path: "secret"
        version: "v2" 
        auth:
            tokenSecretRef:

>               name: ovhcloud-vault-token # The k8s secret that contain your PAT

Reviewer: A quelle etape dans le guide on créer le secret ?
(qui doit se trouver dans LE namespace du SecretManager sinon ça ne fonctionnera pas)

---
title: "Use Kubernetes External Secret Operator with Secret Manager"
excerpt: "Configure External Secret Operator to store Kubernetes secrets on the OVHcloud Secret Manager"
updated: 2025-11-07
---

> [!primary]
> Secret Manager is currently in beta phase. This guide can be updated in the future with the advances of our teams in charge of this product.

## Objective

> This guide explains how set up the Kubernetes External Secret Operator to use the OVHcloud Secret Manager as a provider

Reviewer: ```suggestion
This guide explains how to set up the Kubernetes External Secret Operator to use the OVHcloud Secret Manager as a provider.
```

## Requirements

- An [OVHcloud customer account](/pages/account_and_service_management/account_information/ovhcloud-account-creation).
- Have [ordered an OKMS domain](/pages/manage_and_operate/kms/quick-start) or [created a first secret](/pages/manage_and_operate/secret_manager/secret-manager-ui).

> 

Reviewer: It's mandatory to have a Kubernetes cluster also :)

## Instructions

### Setup the Secret Manager

To allow access to the Secret Manager you will need to create credentials.

> Create an [IAM local user](/pages/account_and_service_management/account_information/ovhcloud-users-management) with acces right on your domain.

Reviewer: Is it possible to do this step though a CLI or TF ?

Because now the user needs to do several clicks and needs to fil several forms in the manager. Not easy at all.
And because we don't help him, the user can do a lot of mistakes.

It's a guide about ESO and secret manager, not IAM so the IAM step needs to be easy :) .

Author: Part wart updated to add a link to IAM documentation.

This will be updated with CLI command and UI steps when available

This user need to have at least the following rights:

- `okms:apikms:secret/create`
- `okms:apikms:secret/version/getData`
- `okms:apiovh:secret/get`

> - `okms:apikms:secret/create`

Reviewer: duplicate line

Then create a Personnal Acces Token (PAT) `user_pat`:

> [!api]
>

> > @api {v1} /me POST /me/identity/user/{user}/token

Reviewer: Possible with a CLI?
Now the user needs to imagine what to do.

Reviewer: Un exemple de reponse valide que l'utilisateur aurait ? :)

Author: Sera actualisé avec les command CLI quand disponble

You will also need the `okms-id` of the OKMS domain you want to use. This ID can be found on the OVHcloud Control Panel.

> ### Setup Sealed Secret

Reviewer: J'avoue je ne comprend pas pourquoi il faut utiliser sealed-secret de bitnami et kubeseal pour utiliser le secret manager.

C'est plus quand on ne veut pas stocker les secrets dans Git et du coup on passe par des sealedsecret.

Sealed Secret allows you to safely store Kubernetes Secrets wherever you want by encrypting them.
This step is optionnal but highly recommendated.

First, install the controller in your cluster.  It will automatically decrypt Sealed Secrets into standard Kubernetes Secrets

```bash
helm repo add sealed-secrets https://bitnami-labs.github.io/sealed-secrets
helm install sealed-secrets -n kube-system sealed-secrets/sealed-secrets
```

Then, install kubeseal cli to encrypt Secrets into Sealed Secrets

```bash
KUBESEAL_VERSION='' # Set this to, for example, KUBESEAL_VERSION='0.23.0'
curl -OL "https://github.com/bitnami-labs/sealed-secrets/releases/download/v${KUBESEAL_VERSION:?}/kubeseal-${KUBESEAL_VERSION:?}-linux-amd64.tar.gz"
tar -xvzf kubeseal-${KUBESEAL_VERSION:?}-linux-amd64.tar.gz kubeseal
sudo install -m 755 kubeseal /usr/local/bin/kubeseal
```

> #### Usage

Reviewer: etant donné qu'aprs dans le guide on applique un SealedSecret, les sections usage et example portent vraiment a confusion.
Donc on peut les supprimer.

- Create your Sealed Secret

```bash
kubeseal -f <secret-file> \
        -w <sealedsecret-output-file> \
        --controller-name=sealed-secrets \
        --controller-namespace=kube-system

kubectl create  -f <sealedsecret-output-file>

# Check if you have access to your original Secret
kubectl get secrets <your-secret-name> -o yaml
```

You can now delete `secret-file` and use `sealedsecret-output-file` instead for a more secure storage

- Delete your Sealed Secret

```bash
kubectl delete sealedsecret <your-sealedsecret-name>
```

#### Example

```bash
$ cat secret.yaml 
apiVersion: v1
kind: Secret
metadata:
  name: secret
type: Opaque
data:

>   value: TVkgQkFTRTY0IEVOQ09ERUQgU0VDUkVUIFZBTFVF

Reviewer: la value correspond a quoi ?

Reviewer: ah c'est un exemple, je pense vraiment que l'exemple porte a confusion, il n'a pas de rapport avec le guide :)

  template:
    metadata:
      name: secret
      namespace: default
    type: Opaque

$ kubectl create -f sealed-secret.yaml 
sealedsecret.bitnami.com/secret created

$ kubectl get secrets secret -o yaml                   
apiVersion: v1
data:
  value: TVkgQkFTRTY0IEVOQ09ERUQgU0VDUkVUIFZBTFVF
kind: Secret
metadata:
  creationTimestamp: "2025-10-13T12:37:25Z"
  name: secret
  namespace: default
  ownerReferences:
  - apiVersion: bitnami.com/v1alpha1
    controller: true
    kind: SealedSecret
    name: secret
    uid: c3a8489f-8125-406b-8a3a-f99b82d432e1
  resourceVersion: "16156798047"
  uid: f3fbfd60-46d7-4211-a9b1-e67d452bc7dc
type: Opaque
```

More information: (<https://github.com/bitnami-labs/sealed-secrets>)

### Setup the Secret Provider in Kubernetes

#### Install the External Secret Operator on your kubernetes

```bash

> helm repo add external-secrets https://charts.external-secrets.io

Reviewer: ```suggestion
helm repo add external-secrets https://charts.external-secrets.io
helm repo update
```

helm install external-secrets \
external-secrets/external-secrets \
-n external-secrets \
--create-namespace \

> --set installCRDs=true

Reviewer: par defaut le champ est a true donc pas obliger de mettre cette ligne installCRDS=true

```

#### Configure External Secret Operator

First, setup a `SecretStore` that is responsible of the synchronization with the Secret Manager.
We configure the SecretStore using HashiCorp Vault with token authentification and with the OKMS endpoint as backend.

Add the `user_pat` as a secret to be able to use it in the charts.

```yaml
---
apiVersion: bitnami.com/v1alpha1
kind: SealedSecret
metadata:

> name: token-secret

Reviewer: ```suggestion
  name: token-secret
```

> namespace: default

Reviewer: ```suggestion
  namespace: default
```

Reviewer: il n'est pas conseillé de mettre des ressources perso dans le namespace default.

spec:
encryptedData:
    token: <user_pat>
template:
    metadata:
    name: token-secret
    namespace: default
    type: Opaque
```

The `SecretStore` resource:

```yaml
apiVersion: external-secrets.io/v1
kind: ClusterSecretStore
metadata:

> name: vault-secret-store

Reviewer: ```suggestion
  name: vault-secret-store
```

spec:

> provider:

Reviewer: le YAML est incorrect, il manque des indentations a partir de ici

    vault:

>     server: "https://{region}.okms.ovh.net/api/<okms_id>" # OKMS endpoint, fill with the correct region and your okms_id 

Reviewer: comment connaitre le nom de sa region alors que dans le manager elle est ecrit comme cela "Europe (France ) Paris" ?
la region il faut l'ecrire en minuscule, majuscule ? Liste predefinie ?

On a un moyen d'obtenir son OKMS endpoint de maniere simple ? :)

    path: "secret"
    version: "v2" 
    auth:
        tokenSecretRef:

>         name: token-secret # The k8s secret that contain your PAT

Reviewer: ```suggestion
          name: token-secret # The k8s secret that contain your PAT
```

>         key: token 

Reviewer: ```suggestion
          key: token 
```

```

> [!info]
> Only [token authentication](https://external-secrets.io/latest/provider/hashicorp-vault/#token-based-authentication) is supported

#### Use External Secret Operator

Once the `SecretStore` is setup you can define `ExternalSecret` that comes from the secret manager.
In the example we use a secret already created on the Secret Manager:

- Path: `prod/database/MySQL`
- Value:
  - `login: admin`
  - `password: my_secret_password`

```yaml
apiVersion: external-secrets.io/v1
kind: ExternalSecret
metadata:

> name: vault-external-secret

Reviewer: il manque les indentations

The user should be a member of a group with the ADMIN role, or if using [IAM policies](/pages/account_and_service_management/account_information/iam-policy-ui) to have at least the following rights on the OKMS domain:

- `okms:apikms:secret/create`
- `okms:apikms:secret/version/getData`
- `okms:apiovh:secret/get`

Then create a Personnal Acces Token (PAT) `user_pat`:

> [!tabs]
> API
>> > [!api]
>> >
>> > @api {v1} /me POST /me/identity/user/{user}/token
>>
>> With the following payload (fill with your values):
>>
>> ```json
>> {
>>   "description": "PAT secret manager for domain xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx",
>>   "name": "pat-secretmanager-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx"
>> }
>> ```
>>
>> API will answer with:
>>
>> ```json
>> {
>>   "creation": "2025-11-13T10:38:44.658926311Z",
>>   "description": "PAT secret manager for domain xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx",
>>   "expiresAt": null,
>>   "lastUsed": null,
>>   "name": "pat-secretmanager-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx",
>>   "token": "eyJhbGciOiJ...punpVAg"
>> }
>> ```
>>
> CLI
>> PAT can also created with the [OVHcloud CLI](https://github.com/ovh/ovhcloud-cli) and the command (fill with your values):
>>
>> ```bash

> >> ovhcloud iam user token create {user} --name pat-secretmanager-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx --description "PAT secret manager for domain xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx"

Reviewer: Or you can create the token AND save it in an environment variable

```
PAT_TOKEN=$(ovhcloud iam user token create {user} --name pat-secretmanager-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx --description "PAT secret manager for domain secretmanager-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx" -j  | jq .details.token |  tr -d '"') ; echo $PAT_TOKEN
```

pod/external-secrets-webhook-7fb59d4b88-9tkl6           0/1     Running   0          12s

NAME                               TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)   AGE
service/external-secrets-webhook   ClusterIP   10.3.43.102   <none>        443/TCP   13s

NAME                                               READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/external-secrets                   1/1     1            1           13s
deployment.apps/external-secrets-cert-controller   0/1     1            0           13s
deployment.apps/external-secrets-webhook           0/1     1            0           13s

NAME                                                          DESIRED   CURRENT   READY   AGE
replicaset.apps/external-secrets-8cbc56569                    1         1         1       13s
replicaset.apps/external-secrets-cert-controller-565fcd479b   1         1         0       13s
replicaset.apps/external-secrets-webhook-7fb59d4b88           1         1         0       13s
```

#### Create a secret containing the PAT

Start by encoding your `user_pat` is base64 so it can be stored in a kubernetes secret.

```bash

> $ echo -n "<token>" | base64

Reviewer: If you saved the token in an env variable (c.f my 1st comment today), you can directly do:

```
PAT_TOKEN_B64=$(echo -n $PAT_TOKEN | base64) ; echo $PAT_TOKEN_B64
```

ZXlKaG...wVkFn
```

> Then create a `secret.yaml`:

Reviewer: Or, easily, create the Secret from the created environment variable (see my comment number 1 and 2 of this review):

```
$ kubectl create secret generic ovhcloud-vault-token -n external-secrets --from-literal=token=$PAT_TOKEN_B64

secret/ovhcloud-vault-token created
```

>> ```
>>
>> As an alternative, it's suggest to store directly PAT in a environment variable:
>>
>> ```bash
>> PAT_TOKEN=$(ovhcloud iam user token create {user} --name pat-secretmanager-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx --description "PAT secret manager for domain secretmanager-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx" -j  | jq .details.token |  tr -d '"') ; echo $PAT_TOKEN
>> ```

Keep safe the value of `token` field as it will never be prompt again and will be used to authenticate on the Secret Manager as `user_pat`.

#### Secret Manager info

You will also need the `region` and the `okms-id` of the OKMS domain you want to use. This ID and this region can be found on the OVHcloud Control Panel.

Or through the [`ovhcloud` CLI](https://github.com/ovh/ovhcloud-cli):

```bash
$ ovhcloud okms list
┌──────────────────────────────────────┬─────────────┐
│ id                                   │ region      │
├──────────────────────────────────────┼─────────────┤
│ xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx │ eu-west-par │
│ xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx │ eu-west-par │
└──────────────────────────────────────┴─────────────┘
```

### Setup the Secret Provider in Kubernetes

> #### Install the External Secret Operator on your kubernetes

Reviewer: ```suggestion
#### Install the External Secret Operator (ESO) on your Kubernetes cluster
```

pod/external-secrets-cert-controller-565fcd479b-xbkcp   0/1     Running   0          12s
pod/external-secrets-webhook-7fb59d4b88-9tkl6           0/1     Running   0          12s

NAME                               TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)   AGE
service/external-secrets-webhook   ClusterIP   10.3.43.102   <none>        443/TCP   13s

NAME                                               READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/external-secrets                   1/1     1            1           13s
deployment.apps/external-secrets-cert-controller   0/1     1            0           13s
deployment.apps/external-secrets-webhook           0/1     1            0           13s

NAME                                                          DESIRED   CURRENT   READY   AGE
replicaset.apps/external-secrets-8cbc56569                    1         1         1       13s
replicaset.apps/external-secrets-cert-controller-565fcd479b   1         1         0       13s
replicaset.apps/external-secrets-webhook-7fb59d4b88           1         1         0       13s
```

#### Create a secret containing the PAT

> Start by encoding your `user_pat` is base64 so it can be stored in a kubernetes secret.

Reviewer: ```suggestion
Start by encoding your `user_pat` is base 64 so it can be stored in a kubernetes secret.
```

```bash
$ echo -n "<token>" | base64
ZXlKaG...wVkFn
```

Or if `user_pat` was store in a environment variable:

```bash
PAT_TOKEN_B64=$(echo -n $PAT_TOKEN | base64) ; echo $PAT_TOKEN_B64
```

> Then create a `secret.yaml`:

Reviewer: ```suggestion
Then create a `secret.yaml` file:
```

```yaml
 apiVersion: v1
kind: Secret
metadata:
  name: ovhcloud-vault-token
  namespace: external-secrets
data:
  token: ZXlKaG...wVkFn
```

> 

Reviewer: ```suggestion

And apply it with the `kubectl apply -f secret.yaml` command.

```

Or if using an environment variable:

```bash
kubectl create secret generic ovhcloud-vault-token -n external-secrets --from-literal=token=$PAT_TOKEN_B64

> 

Reviewer: ```suggestion
```

> secret/ovhcloud-vault-token created

Reviewer: ```suggestion
```

```

> And apply the ressource to the cluster:

Reviewer: ```suggestion
```

> ```bash

Reviewer: ```suggestion
```

> kubectl apply -f secret.yaml

Reviewer: ```suggestion
```

> ```

Reviewer: ```suggestion
```

The secret should have been created:

```bash
$ kubectl get secret ovhcloud-vault-token -n external-secrets
NAME                   TYPE     DATA   AGE
ovhcloud-vault-token   Opaque   1      5m
```

#### Configure External Secret Operator

First, setup a `ClusterSecretStore` that is responsible of the synchronization with the Secret Manager.
We configure the SecretStore using HashiCorp Vault with token authentification and with the OKMS endpoint as backend.

Add the `user_pat` as a secret to be able to use it in the charts.

> To define a new `ClusterSecretStore` resource, create a `clustersecretstore.yaml` file with the followong content:

Reviewer: ```suggestion
To define a new `ClusterSecretStore` resource, create a `clustersecretstore.yaml` file with the following content:
```
